### PR TITLE
Use Redis++ ConnectionObject to set TCP socket timeout

### PR DESCRIPTION
**Pull request by billschereriii**
TCP default timeout is ridiculously long, so if we attempt to connect to a missing database, we're stuck waiting on the TCP timeout repeatedly. This will improve our reponsiveness in such cases

### CODE REVIEW

**Comment by billschereriii** [comment]
Hmm.  A Redis connection based on the ConnectionOptions object doesn't seem to be stable. Will investigate further next week

File: src/cpp/redis.cpp [hunk 1]
```diff
@@ -722,10 +722,28 @@ inline void Redis::_add_to_address_map(SRAddress& db_address)
 
 inline void Redis::_connect(SRAddress& db_address)
 {
+    // Build a connections object for this connection
+    // No need to repeat the build on each connection attempt
+    // so we do it outside the loop
+    sw::redis::ConnectionOptions connectOpts;
+    if (db_address._is_tcp) {
+        connectOpts.host = db_address._tcp_host;
+        std::cout << "Connecting to host: " + connectOpts.host + "; ";
```
Comment (ashao) [inline src/cpp/redis.cpp:731]: Can we move this into the logger somehow instead of outputting directly to STDOUT

File: src/cpp/redis.cpp [hunk 2]
```diff
@@ -722,10 +722,28 @@ inline void Redis::_add_to_address_map(SRAddress& db_address)
...
+        connectOpts.port = db_address._tcp_port;
+        std::cout << "Connecting to port: " + std::to_string(connectOpts.port) + "\n";
```
Comment (ashao) [inline src/cpp/redis.cpp:733]: Same

File: src/cpp/rediscluster.cpp [hunk 3]
```diff
@@ -1075,6 +1075,24 @@ inline CommandReply RedisCluster::_run(const Command& cmd, std::string db_prefix
 // Connect to the cluster at the address and port
 inline void RedisCluster::_connect(SRAddress& db_address)
 {
+    // Build a connections object for this connection
+    // No need to repeat the build on each connection attempt
+    // so we do it outside the loop
+    sw::redis::ConnectionOptions connectOpts;
+    if (db_address._is_tcp) {
+        connectOpts.host = db_address._tcp_host;
+        std::cout << "Connecting to host: " + connectOpts.host + "; ";
```
Comment (ashao) [inline src/cpp/rediscluster.cpp:1084]: Also here capture to logger, or eliminate the line

File: src/cpp/redis.cpp [hunk 4]
```diff
@@ -722,10 +722,28 @@ inline void Redis::_add_to_address_map(SRAddress& db_address)
...
+        connectOpts.type = sw::redis::ConnectionType::TCP;
+    }
+    else {
+        connectOpts.path = db_address._uds_file;
+        connectOpts.type = sw::redis::ConnectionType::UNIX;
+    }
+    connectOpts.socket_timeout = std::chrono::milliseconds(250);
```
Comment (ashao) [inline src/cpp/redis.cpp:740]: Let's make the 250 a #define somewhere

File: src/cpp/rediscluster.cpp [hunk 5]
```diff
@@ -1075,6 +1075,24 @@ inline CommandReply RedisCluster::_run(const Command& cmd, std::string db_prefix
...
+        connectOpts.port = db_address._tcp_port;
+        std::cout << "Connecting to port: " + std::to_string(connectOpts.port) + "\n";
+        connectOpts.type = sw::redis::ConnectionType::TCP;
+    }
+    else {
+        throw SRInternalException(
+            "RedisCluster encountered a UDS request in _connect()");
+    }
+    connectOpts.socket_timeout = std::chrono::milliseconds(250);
```
Comment (ashao) [inline src/cpp/rediscluster.cpp:1093]: Same suggestion to move htis to a #define

**Review by ashao — CHANGES_REQUESTED** [review]

File: src/cpp/redis.cpp (hunk 1 above)
Comment (billschereriii) [reply src/cpp/redis.cpp:731]: Debug text -- will remove

File: src/cpp/redis.cpp (hunk 2 above)
Comment (billschereriii) [reply src/cpp/redis.cpp:733]: same

File: src/cpp/rediscluster.cpp (hunk 3 above)
Comment (billschereriii) [reply src/cpp/rediscluster.cpp]: same

File: src/cpp/redis.cpp (hunk 4 above)
Comment (billschereriii) [reply src/cpp/redis.cpp]: done -- good thought

**Review by ashao — APPROVED** [review]

**Pull request merged**